Mathlib/FieldTheory/PurelyInseparable/Tower.lean

```diff
@@ -153,6 +153,30 @@ theorem sepDegree_mul_sepDegree_of_isAlgebraic (K : Type v) [Field K] [Algebra F
     sepDegree F E * sepDegree E K = sepDegree F K := by
   simpa only [Cardinal.lift_id] using lift_sepDegree_mul_lift_sepDegree_of_isAlgebraic F E K
 
+@[stacks 09HK "Part 2, `finInsepDegree` variant for finite extensions"]
+lemma finInsepDegree_mul_finInsepDegree_of_finite [Module.Finite F K] :
+    finInsepDegree F E * finInsepDegree E K = finInsepDegree F K := by
+  have : Module.Finite E K := .of_restrictScalars_finite F _ _
+  apply mul_right_injective₀ (NeZero.ne (finSepDegree F K))
+  dsimp only
+  rw [Field.finSepDegree_mul_finInsepDegree,
+    ← Field.finSepDegree_mul_finSepDegree_of_isAlgebraic F E K,
+    mul_mul_mul_comm, Field.finSepDegree_mul_finInsepDegree, Field.finSepDegree_mul_finInsepDegree,
+    Module.finrank_mul_finrank]
```

Comment:
How far are we from the statement for arbitrary algebraic extensions? Is it fine if we add this for the time being?

Reply:
Do you have another proof for it? In my original proof, linearly disjoint is needed for arbitrary algebraic extension case. Now linearly disjoint is added to mathlib, so it should be doable now.

Reply:
Now I'm working on the tower law of inseparable degree, it should be done in one or two days.

Reply:
PR created as #24806. Comments welcome. Maybe you want to depend on that PR.

By the way, now `Mathlib/FieldTheory/PurelyInseparable/Tower.lean` imports linearly disjoint file, which in turn imports algebraic independent file. I hope this will not introduce circular import in your PR.